File: src/JavaScript/easy/array/problem349.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */

var intersection = function(nums1, nums2) {
    let unique_list = []
    nums1.forEach(num => {
        if(nums2.includes(num)){
            unique_list.push(num)
        }
    });
    return Array.from(new Set(unique_list))
};


// let nums1 = [1,2,2,1], nums2 = [2,2]
let nums1 = [4,9,5], nums2 = [9,4,9,8,4]

intersection(nums1, nums2)

File: src/Python/easy/array/problem349.py
from typing import List

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        unique_list = []
        for n in nums1:
            if n in nums2:
                unique_list.append(n)
        return list(set(unique_list))



nums1 = [1,2,2,1]; nums2 = [2,2]

solution = Solution()
solution.intersection(nums1, nums2)